server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -201,29 +200,27 @@ public AsyncJobStatus queueJob(JobBuilder builder) {
         }
 
         AsyncJobStatus job = this.buildJobStatus(builder);
+        job.setState(JobState.QUEUED);
+
+        // Persist the job status so that the ID will be generated.
+        job = this.jobCurator.create(job);
```

Comment:
@Ceiu We already chatted about this but wanted to leave notes so I don't forget:

I think we have to do it this way because once the message is sent, it is possible that the message would be received just before the actual status has been persisted, potentially leading to the job not being found once received.

We could create the initial AsyncJobStatus with a state of "CREATED" and then set it to queued once the message is sent successfully.

In the Quartz job system, candlepin generates the job's id. Any reason we moved away from that approach? We wouldn't have to pre-persist the status this way.

Reply:
> I think we have to do it this way because once the message is sent, it is possible that the message would be received just before the actual status has been persisted, potentially leading to the job not being found once received.

Rog thinks this is going to be a problem regardless; the transaction needs to end before another node will see it in the DB.

Reply:
> In the Quartz job system, candlepin generates the job's id. Any reason we moved away from that approach? We wouldn't have to pre-persist the status this way.

Rog:  "I think I was torn between the two, forgetting about this part. You can see the setId method still commented out. In light of this, I would lean the other way again"

Reply:
I've added a card to come up with a better way of handling this [1]. For now, I'll create the initial status with the state of CREATED, update the state to QUEUED once the message is sent and attempt to delete on message send failure.

[1] https://projects.engineering.redhat.com/browse/ENT-1095

Reply:
DONE